ang/test/civicase/ActivitiesCalendar.spec.js

```diff
@@ -2,20 +2,63 @@
 
 (function ($, _, moment) {
   describe('civicaseActivitiesCalendarController', function () {
-    var $controller, $scope, $rootScope, activitiesMockData,
-      dates, formatActivity, mockCaseId;
+    var $controller, $q, $scope, $rootScope, crmApi, dates;
 
-    beforeEach(module('civicase', 'civicase.data', 'ui.bootstrap'));
+    beforeEach(module('civicase', 'crmUtil', 'civicase.data', 'ui.bootstrap'));
 
-    beforeEach(inject(function (_$controller_, _$rootScope_,
-      _activitiesMockData_, datesMockData, _formatActivity_) {
+    beforeEach(inject(function (_$controller_, _$q_, _$rootScope_, _crmApi_,
+      datesMockData) {
       $controller = _$controller_;
+      $q = _$q_;
       $rootScope = _$rootScope_;
-      activitiesMockData = _activitiesMockData_.get();
+      crmApi = _crmApi_;
+
+      $scope = $rootScope.$new();
       dates = datesMockData;
-      formatActivity = _formatActivity_;
+
+      crmApi.and.returnValue($q.resolve({ values: [] }));
     }));
 
+    describe('when uib-datepicker signals that it is ready', function () {
+      beforeEach(function () {
+        spyOn($scope, '$emit').and.callThrough();
+
+        initController();
+        $rootScope.$emit('uibDaypicker::compiled');
+      });
+
+      it('starts loading the days with incomplete activities', function () {
+        expect(crmApi).toHaveBeenCalledWith('Activity', 'getdayswithactivities', {
+          case_id: $scope.caseId,
+          status_id: { 'IN': CRM.civicase.activityStatusTypes.incomplete }
+        });
+      });
+
+      it('does not load the days with complete activities right away', function () {
+        expect(crmApi.calls.count()).toBe(1);
+      });
+
+      describe('when loading is complete', function () {
+        beforeEach(function () {
+          crmApi.calls.reset();
```

Comment:
Why are we calling `crmApi.calls.reset()`? How will this make sure it will call the next then statement. Just wanted to understand this a bit more.

Reply:
Because I want to make sure that jasmine ignore the previous call to `crmApi` (the one to fetch the days with incomplete activities) before making another assertion on whether the function has been called or not.

> How will this make sure it will call the next then statement.

`calls.reset()` has no influence on whether the spy function will be called or not, it's simply a way to tell jasmine to ignore all the previous calls made prior, see here https://jasmine.github.io/2.0/introduction.html#section-32

Reply:
Thanks for the info. It was helpful.